sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -390,13 +393,21 @@ class ProjectOverviewView extends VerticalLayout{
         }
         else if (sampleCount.passingSamples == sampleCount.totalSampleCount) {
             return State.COMPLETED
-        }
-        else if (sampleCount.passingSamples < sampleCount.totalSampleCount) {
+        } else if (sampleCount.passingSamples < sampleCount.totalSampleCount) {
             return State.IN_PROGRESS
-        }
-        else {
+        } else {
             //unexpected!!
             throw new IllegalStateException("status count $sampleCount.passingSamples must not be greater total count $sampleCount.totalSampleCount")
         }
     }
+
+    private void setupFilters(ListDataProvider<ProjectSummary> projectSummaryListDataProvider) {
+        HeaderRow customerFilterRow = projectGrid.appendHeaderRow()
+        GridUtils.setupColumnFilter(projectSummaryListDataProvider,
+                projectGrid.getColumn("ProjectCode"),
+                customerFilterRow)
+        GridUtils.setupColumnFilter(projectSummaryListDataProvider,
+                projectGrid.getColumn("ProjectTitle"),
+                customerFilterRow)
+    }
```

Comment:
I like that you moved that to its own method. The current implementation has a slight problem though. The method is called every time the data provider is reloaded (that is correct IMO), however every time a new header row is added. This can be solved by counting the header rows and, in case the count is greater than 1, assigning this row as the filter row. 
You can reproduce the behaviour by subscribing/unsubscribing from a project.

Reply:
Nice catch 👁️ I updated the code with your suggestion